### DISCUSSION TITLE
I don't seem to be able to edit atomFamily atom from write atom

### DISCUSSION THREAD

**Discussion by pencil-user** [pmndrs/jotai#1162] [Q&A] [answered]
I am using Jotai 1.6.6 and Typescript 4.2.3. I am trying to edit atom in stringFamily from Write Atom instead of from the hook. But this shows an error:

```
const stringFamily = atomFamily<string, Atom<string>>(() => atom('foo'))

const writeString = atom(null, (get, set, params:{ id: string, content: string }) => {
   set(stringFamily(params.id), params.content)
})
```
the red underline is on `stringFamily(params.id)` part and it reads

```
Argument of type 'Atom<string>' is not assignable to parameter of type 'WritableAtom<string, string, void | Promise<void>>'.
  Property 'write' is missing in type 'Atom<string>' but required in type '{ write: Write<string, void | Promise<void>>; onMount?: OnMount<string, void | Promise<void>> | undefined; }'.ts(2345)
```

(Code where I was trying to do this is actually much longer, but I wrote this example for the purposes of demonstration). Not sure what to do here. Am I doing something wrong?

**Comment by dai-shi** [comment, marked as answer]
Hi, is this TypeScript only issue? If so, please check if you have `"strict": true` in tsconfig.json. jotai requires `strictNullChecks`.

**Reply by pencil-user** [reply]
First of all, thanks for quick answer, @dai-shi ! I do appreciate it.

I don't know whether it is TypeScript only issue because I only ever use TypeScript.

I do have `"strict" : true`, so that's not the problem. I've also tried to update to newest TypeScript version with no change. The problem seems to be that the types I get from atomFamily don't quite match the types from atom. I tried this:

```
const stringFamily = atomFamily<string, Atom<string>>(() => atom('foo'))

const stringAtom = atom('foo')

const writeString = atom(null, (get, set, params:{ id: string, content: string }) => {
   const myAtom = stringFamily(params.id)
   set(stringFamily(params.id), params.content)
})
```

It doesn't work, but what is interesting is that vsCode shows that `stringAtom` and `myAtom` are of subtly different types. `stringAtom`:
```
const stringAtom: Atom<string> & {
    write: Write<SetStateAction<string>, void>;
    onMount?: OnMount<SetStateAction<string>, void> | undefined;
} & WithInitialValue<...>
```
myAtom:
```
const myAtom: Atom<string>
```

Maybe a lack of `write` is what prevents it from working?

**Reply by dai-shi** [reply]
Oops, sorry, it's not about strictNullChecks.
It's because of explicit/narrower typing of your family.
You should do this instead:
```js
const stringFamily = atomFamily((_str: string) => atom('foo'))
```

**Reply by pencil-user** [reply]
Much better, thanks!
